ddtrace.gemspec

```diff
@@ -36,21 +39,12 @@ Gem::Specification.new do |spec|
       ['.dockerignore', '.env', '.gitattributes', '.gitlab-ci.yml', '.rspec', '.rubocop.yml',
        '.rubocop_todo.yml', '.simplecov', 'Appraisals', 'Gemfile', 'Rakefile', 'docker-compose.yml', '.pryrc', '.yardopts'].include?(f)
     end
+  # rubocop:enable all
   spec.executables   = ['ddtracerb']
   spec.require_paths = ['lib']
 
-  # Important note: This `if` ONLY works for development. When packaging up the gem, Ruby runs this code, and hardcodes
-  # the output in the `.gem` file that gets uploaded to rubygems. So the decision here gets hardcoded, we don't actually
-  # pick a version depending on the Ruby customers are running, as it may appear.
-  # For more context, see the discussion in
-  # * https://github.com/DataDog/dd-trace-rb/pull/1739
-  # * https://github.com/DataDog/dd-trace-rb/pull/1336
-  if RUBY_VERSION >= '2.2.0'
-    spec.add_dependency 'msgpack'
-  else
-    # msgpack 1.4 fails for Ruby 2.1: https://github.com/msgpack/msgpack-ruby/issues/205
-    spec.add_dependency 'msgpack', '< 1.4'
-  end
```

Comment:
I cleaned this up because it simply never worked in first place.

Reply:
Would you consider leaving in a comment along the lines of...

```
# msgpack 1.4 fails for Ruby 2.1 (see https://github.com/msgpack/msgpack-ruby/issues/205)
# so a restriction needs to be manually added for the `Gemfile`.
#
# We can't add a restriction here, since there's no way to add it only for older
# rubies, see #1739 and #1336 for an extended discussion about this
```

This way it's clear that there's a limitation here and why; otherwise the next time someone bumps into the issue they may not know that we've already tried to solve this and may lose a bunch of time rediscovering it.